### PR TITLE
Add specfics to ug

### LINKED ISSUE #288: [PE-D][Tester C] Role Validation after reading from JSON
**How to reproduce:** Edit the role name in JSON file

**Expected:** The role name should be validated when reading the JSON file, then convert to the most similar role name or just remove the person, or alert the user.

**Actual:** There is no validation and the invalid role name is shown

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: purivirakarin/ped#11

### PR DESCRIPTION
**Pull request by jannaleong**
Fixes #288 #275 #269

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -24,44 +24,42 @@ Get started by following the steps in this user guide! Experience the difference
...
-PoochPlanner is your go-to address book, tailor-made for dog cafe owners like you. With four main contact types—**person, staff, supplier, maintainer**—each equipped with its own set of attributes and functions, managing your contacts has never been easier.
+PoochPlanner is your go-to address book, tailor-made for dog cafe owners like you. With four main contact types — **person, staff, supplier, maintainer** — each equipped with its own set of attributes and functions, managing your contacts has never been easier.
 
 Contact | Definition and Examples
 --------|------------------
-Person | Any persons that do not hold a special relationship.<br> E.g. Neighbours, Work Friends.
-Staff | Any full-time or part-time employees that do not do any specialised work.<br> E.g. Waiters, Waitresses, Dishwashers.
-Supplier | Any logistics partner that supplies stock for your dog cafe.<br> E.g. Kibbles Supplier, Coffee Beans Supplier.
-Maintainer | Any externally hired specialised persons who take care of the dogs.<br> E.g. Dog Groomers, Vet.
+Person | Any persons that do not hold a special relationship.<br> E.g. Neighbours, Work friends.
```
Comment (jamessinmaojun) [inline docs/UserGuide.md:31]: Standardise capitalisation. e.g. Work Friends or work friends. same for Kibbles Supplier, Coffee Beans Supplier, Dog Groomers in the lines below.

My take is not to capitalise as these are not proper nouns.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -70,52 +68,56 @@ While exploring PoochPlanner’s features with this user guide, you can take not
...
-Case-Sensitive | The casing of the alphabetic characters matters (e.g. “good” is different from “GOOD”).
-Case-Insensitive | The casing of the alphabetic characters does not matter (e.g. “good” is taken to be equal to “GOOD”).
+Prefix | Prefixes are like fields in a form you are required to fill up. They capture important information that is needed to execute the command. View PoochPlanner’s [Prefix Summary](#prefix-summary).
+Case-sensitive | The casing of the alphabetic characters matters (e.g. “good” is different from “GOOD”).
+Case-insensitive | The casing of the alphabetic characters does not matter (e.g. “good” is taken to be equal to “GOOD”).
+Space-sensitive | The number of spaces in a command matters (e.g. “happy puppy” is different from “happypuppy”).
+Space-insensitive | The number of spaces in a command does not matter (e.g. “happy puppy” is taken to be equal to “happypuppy”).
+Delimeter | A seperator in the command input that defines the boundaries between information in a command. ";" is a delimeter for our commands.
```
Comment (jamessinmaojun) [inline docs/UserGuide.md:88]: Spelling error, should be _delimiter_

File: docs/UserGuide.md [hunk 3]
```diff
@@ -70,52 +68,56 @@ While exploring PoochPlanner’s features with this user guide, you can take not
...
+Contact card | The box displayed for each contact, containing all their details.
 
 
 ## Quick start
 
-1. Ensure you have Java `11` installed in your Computer.
+1. Ensure you have Java `11` installed in your computer.
```
Comment (jamessinmaojun) [inline docs/UserGuide.md:94]: "installed _on_" is more commonly used than "installed _in_"

File: docs/UserGuide.md [hunk 4]
```diff
@@ -70,52 +68,56 @@ While exploring PoochPlanner’s features with this user guide, you can take not
...
 
-1. Type the command in the command box and press Enter to execute it. 
+1. Type the command in the input box and press enter to execute it. 
    Some example commands you can try:
 
-   * `/add-person ; name : Janna  ; phone : 98765435 ; address : Poochie Street 24 ; email : ihelppooches@gmail.com` : Adds a general contact named `Janna` to the Pooch Planner.
+   * `/add-person ; name : Janna  ; phone : 98765435 ; address : Poochie Street 24 ; email : ihelppooches@gmail.com` : Adds a general contact named `Janna` to the PoochPlanner.
 
-   * `/delete ; name : Janna` : Deletes the contact named Janna from Pooch Planner.
+   * `/delete ; name : Janna` : Deletes the contact named Janna from PoochPlanner.
```
Comment (jamessinmaojun) [inline docs/UserGuide.md:109]: similar to above, add "your" identifier

**Review by yleeyilin — APPROVED** [review]
lgtm!

File: docs/UserGuide.md [hunk 5]
```diff
@@ -181,7 +185,7 @@ The table below summarises the add command, format and examples for each contact
 
 * Name is case-insensitive but space-sensitive.<br>
 
-* Email must in format {example}@{domain} or {example}@{domain}.{extension}
+* Email must in format {example}@{domain} or {example}@{domain}.{extension} <br>
```
Comment (jamessinmaojun) [inline docs/UserGuide.md:188]: grammar error, "Email must **be** in **the** format..."

File: docs/UserGuide.md [hunk 6]
```diff
@@ -181,7 +185,7 @@ The table below summarises the add command, format and examples for each contact
...
 
 * Phone number must be in format {Number} and at least 3 digits long.<br>
```
Comment (jamessinmaojun) [inline docs/UserGuide.md:190]: grammar error, "Phone number must be in **the** format..."

File: docs/UserGuide.md [hunk 7]
```diff
@@ -190,6 +194,7 @@ The table below summarises the add command, format and examples for each contact
 * Salary and commission must be in format ${Number}/hr.<br>
 
 * Price must be in format ${Number}/{quantity}.<br>
+* Employment must be either part-time or full-time.<br>
```
Comment (jamessinmaojun) [inline docs/UserGuide.md:197]: good to put "part-time" and "full-time" in quotes to indicate that these are the exact values that `employment` accepts.

File: docs/UserGuide.md [hunk 8]
```diff
@@ -306,12 +315,13 @@ Examples: <br>
...
 
 * You can use `/list` to see the full list again after searching for a contact! <br>
+* For your convenience, if there are repeated fields, only the value of the latest repeated field will be processed. So, if you have mistakenly entered the wrong value, simply retype the field and value in the same line.<br>
 
 </div>
 
 #### Deleting a contact : `delete`
 
-Deletes a **person / staff / supplier / maintainer** contact from the Pooch Planner so that you can remove outdated contacts.
+Deletes a **person / staff / supplier / maintainer** contact from the PoochPlanner so that you can remove outdated contacts.
```
Comment (jamessinmaojun) [inline docs/UserGuide.md:324]: ...from _your_ PoochPlanner...

File: docs/UserGuide.md [hunk 9]
```diff
@@ -338,14 +350,14 @@ Example: <br>
...
-* Name must be present in Pooch Planner.<br>
+* Name must be present in PoochPlanner.<br>
 </div>
 
 ### Special Features
 
 #### Rating a Contact : `rate`
 
-Gives a **person / staff / supplier / maintainer** contact from the Pooch Planner a performance rating so that cafe owners can track the performance of their contacts.
+Gives a **person / staff / supplier / maintainer** contact from the PoochPlanner a performance rating so that cafe owners can track the performance of their contacts.
```
Comment (jamessinmaojun) [inline docs/UserGuide.md:360]: ...from _your_ PoochPlanner...

File: docs/UserGuide.md [hunk 10]
```diff
@@ -388,7 +402,7 @@ Example:
 
 #### Pinning a contact : `pin`
 
-Pins the specified contact on Pooch Planner so that your important contacts will consistently appear at the top on the contact list.
+Pins the specified contact on PoochPlanner so that your important contacts will consistently appear at the top on the contact list.
```
Comment (jamessinmaojun) [inline docs/UserGuide.md:405]: change to "...at the top ~~on the~~ _of your_ contact list."

File: docs/UserGuide.md [hunk 11]
```diff
@@ -416,22 +432,23 @@ Example: <br>
 
 * Name is a compulsory field that is case-insensitive but space-sensitive.<br>
 
-* Name must be present in Pooch Planner.<br>
+* Name must be present in PoochPlanner.<br>
 
-* Using pin command on a contact that has been pinned do not make any changes to Pooch Planner.<br>
+* Using pin command on a contact that has been pinned do not make any changes to PoochPlanner.<br>
```
Comment (jamessinmaojun) [inline docs/UserGuide.md:437]: grammar errors:
1. "Using _the_ pin command..."
2.  "...~~do~~ _does_ not make any changes..."

File: docs/UserGuide.md [hunk 12]
```diff
@@ -416,22 +432,23 @@ Example: <br>
...
 
 * Use the pin command for your frequently contacted addresses! <br>
+* For your convenience, if there are repeated fields, only the value of the latest repeated field will be processed. So, if you have mistakenly entered the wrong value, simply retype the field and value in the same line.<br>
 
 </div>
 
 #### Unpinning a contact : `unpin`
 
-Unpins the specified contact on Pooch Planner so that your less important contacts can be removed from the top on the contact list.
+Unpins the specified contact on PoochPlanner so that your less important contacts can be removed from the top on the contact list.
```
Comment (jamessinmaojun) [inline docs/UserGuide.md:451]: Same here, change to "...removed from the top ~~on the~~ _of your_ contact list."

File: docs/UserGuide.md [hunk 13]
```diff
@@ -458,13 +477,19 @@ Example: <br>
 
 * Name is a compulsory field that is case-insensitive but space-sensitive.<br>
 
-* Name must be present in Pooch Planner.<br>
+* Name must be present in PoochPlanner.<br>
 
-* Using unpin command on a contact that has been unpinned do not make any changes to Pooch Planner.<br>
+* Using unpin command on a contact that has been unpinned do not make any changes to PoochPlanner.<br>
```
Comment (jamessinmaojun) [inline docs/UserGuide.md:482]: Same grammatical errors here:
1. missing article "the", change to "Using _the_ unpin command..."
2. "~~do~~ _does_ not make any changes..."

File: docs/UserGuide.md [hunk 14]
```diff
@@ -540,16 +569,20 @@ Examples: <br>
 <div markdown="span" class="alert alert-warning">:exclamation: **Constraints:**<br>
 
 * Name and Note are compulsory fields.<br>
-* Name is case-insensitive but space-sensitive <br>
-* Name must be present in Pooch Planner.<br>
-* Note cannot be empty 
+* Deadlines are optional fields. <br>
+* Deadlines must follow "YYYY-MM-DD" format. <br>
+* Name is case-insensitive but space-sensitive. <br>
```
Comment (jamessinmaojun) [inline docs/UserGuide.md:574]: Change "name" to plural form to standardise with "deadlines",

i.e. "Names are case-insensitive but space-sensitive."

Same for the following 2 lines

**Review by jamessinmaojun — APPROVED** [review]
Few minor errors, otherwise lgtm!

File: docs/UserGuide.md (hunk 1 above)
Comment (jannaleong) [reply docs/UserGuide.md:31]: noted !! will change to remove capitalisation

File: docs/UserGuide.md (hunk 2 above)
Comment (jannaleong) [reply docs/UserGuide.md:88]: changed :)

File: docs/UserGuide.md (hunk 3 above)
Comment (jannaleong) [reply docs/UserGuide.md:94]: changed

File: docs/UserGuide.md [hunk 15]
```diff
@@ -70,52 +68,56 @@ While exploring PoochPlanner’s features with this user guide, you can take not
...
-   A GUI similar to the below should appear in a few seconds. Note that the app does not contain sample data.<br>
+   A GUI similar to the one below should appear in a few seconds. Note that the app does not contain sample data.<br>
    ![Ui](images/Ui.png)
 
-1. Type the command in the command box and press Enter to execute it. 
+1. Type the command in the input box and press enter to execute it. 
    Some example commands you can try:
 
-   * `/add-person ; name : Janna  ; phone : 98765435 ; address : Poochie Street 24 ; email : ihelppooches@gmail.com` : Adds a general contact named `Janna` to the Pooch Planner.
+   * `/add-person ; name : Janna  ; phone : 98765435 ; address : Poochie Street 24 ; email : ihelppooches@gmail.com` : Adds a general contact named `Janna` to the PoochPlanner.
```
[the remark this answers is not in the document]
Comment (jannaleong) [reply docs/UserGuide.md:107]: changed

File: docs/UserGuide.md (hunk 4 above)
Comment (jannaleong) [reply docs/UserGuide.md:109]: changed

File: docs/UserGuide.md (hunk 6 above)
Comment (jannaleong) [reply docs/UserGuide.md:190]: changed

File: docs/UserGuide.md (hunk 7 above)
Comment (jannaleong) [reply docs/UserGuide.md:197]: changed

File: docs/UserGuide.md (hunk 9 above)
Comment (jannaleong) [reply docs/UserGuide.md:360]: changed

File: docs/UserGuide.md (hunk 10 above)
Comment (jannaleong) [reply docs/UserGuide.md:405]: changed

File: docs/UserGuide.md (hunk 11 above)
Comment (jannaleong) [reply docs/UserGuide.md:437]: changed

File: docs/UserGuide.md (hunk 12 above)
Comment (jannaleong) [reply docs/UserGuide.md:451]: changed

File: docs/UserGuide.md (hunk 13 above)
Comment (jannaleong) [reply docs/UserGuide.md:482]: changed

File: docs/UserGuide.md (hunk 14 above)
Comment (jannaleong) [reply docs/UserGuide.md:574]: changed

**Review by jamessinmaojun — APPROVED** [review]
lgtm

**Pull request merged**